float64 inner # [m]
float64 outer # [m]
